src/Client.php

```diff
@@ -56,21 +57,60 @@ final class Client implements ClientInterface
      */
     private $integrations;
 
+    /**
+     * @var SerializerInterface|null The serializer of the client
+     */
+    private $serializer;
+
+    /**
+     * @var RepresentationSerializerInterface|null The representation serializer of the client
+     */
+    private $representationSerializer;
+
+    /**
+     * @var StacktraceBuilder
+     */
+    private $stacktraceBuilder;
+
+    /**
+     * @var string The Sentry SDK identifier
+     */
+    private $sdkIdentifier;
+
+    /**
+     * @var string the SDK version of the Client
+     */
+    private $sdkVersion;
+
     /**
      * Constructor.
      *
-     * @param Options               $options      The client configuration
-     * @param TransportInterface    $transport    The transport
-     * @param EventFactoryInterface $eventFactory The factory for events
-     * @param LoggerInterface|null  $logger       The PSR-3 logger
+     * @param Options                                $options                  The client configuration
+     * @param TransportInterface                     $transport                The transport
+     * @param string|null                            $sdkIdentifier            The Sentry SDK identifier
+     * @param string|null                            $sdkVersion               The Sentry SDK version
+     * @param SerializerInterface|null               $serializer               The serializer
+     * @param RepresentationSerializerInterface|null $representationSerializer The serializer for function arguments
+     * @param LoggerInterface|null                   $logger                   The PSR-3 logger
      */
-    public function __construct(Options $options, TransportInterface $transport, EventFactoryInterface $eventFactory, ?LoggerInterface $logger = null)
-    {
+    public function __construct(
+        Options $options,
+        TransportInterface $transport,
+        ?string $sdkIdentifier = null,
+        ?string $sdkVersion = null,
+        ?SerializerInterface $serializer = null,
+        ?RepresentationSerializerInterface $representationSerializer = null,
+        ?LoggerInterface $logger = null
+    ) {
         $this->options = $options;
         $this->transport = $transport;
-        $this->eventFactory = $eventFactory;
         $this->logger = $logger ?? new NullLogger();
         $this->integrations = IntegrationRegistry::getInstance()->setupIntegrations($options, $this->logger);
+        $this->serializer = $serializer ?? new Serializer($this->options);
+        $this->representationSerializer = $representationSerializer ?? new RepresentationSerializer($this->options);
+        $this->stacktraceBuilder = new StacktraceBuilder($options, $this->representationSerializer);
+        $this->sdkIdentifier = $sdkIdentifier ?? self::SDK_IDENTIFIER;
+        $this->sdkVersion = $sdkVersion ?? PrettyVersions::getVersion('sentry/sentry')->getPrettyVersion();
```

Comment:
We should be able to use `Versions::rootPackageName()` in place of hardcoding the package name